pkg/cmd/cli/cmd/rsync/rsync.go

```diff
@@ -0,0 +1,218 @@
+package rsync
+
+import (
+	"errors"
+	"fmt"
+	"io"
+
+	"github.com/golang/glog"
+	"github.com/spf13/cobra"
+	kcmdutil "k8s.io/kubernetes/pkg/kubectl/cmd/util"
+
+	"github.com/openshift/origin/pkg/cmd/util/clientcmd"
+)
+
+const (
+	// RsyncRecommendedName is the recommended name for the rsync command
+	RsyncRecommendedName = "rsync"
+
+	rsyncLong = `
+Copy local files to a container
+
+This command will attempt to copy local files to a remote container. It will default to the
+first container if none is specified, and will attempt to use 'rsync' if available locally and in the
+container. If 'rsync' is not present, it will attempt to use 'tar' to send files to the container.`
+
+	rsyncExample = `
+  # Synchronize a local directory with a pod directory
+  $ %[1]s ./local/dir/ POD:/remote/dir`
+
+	noRsyncUnixWarning    = "rsync command not found in path. Please use your package manager to install it."
+	noRsyncWindowsWarning = "rsync command not found in path. Download cwRsync for windows and add it to your path."
+)
+
+// copyStrategy
+type copyStrategy interface {
+	Copy(source, destination *pathSpec, out, errOut io.Writer) error
+	Validate() error
+	String() string
+}
+
+// executor executes commands
+type executor interface {
+	Execute(command []string, in io.Reader, out, err io.Writer) error
+}
+
+// forwarder forwards pod ports to the local machine
+type forwarder interface {
+	ForwardPorts(ports []string, stopChan <-chan struct{}) error
+}
+
+// RsyncOptions holds the options to execute the sync command
+type RsyncOptions struct {
+	Namespace     string
+	ContainerName string
+	Source        *pathSpec
+	Destination   *pathSpec
+	Strategy      copyStrategy
+	StrategyName  string
+	Quiet         bool
+	Delete        bool
+
+	Out    io.Writer
+	ErrOut io.Writer
+}
+
+// NewCmdRsync creates a new sync command
+func NewCmdRsync(name, parent string, f *clientcmd.Factory, out, errOut io.Writer) *cobra.Command {
+	o := RsyncOptions{
+		Out:    out,
+		ErrOut: errOut,
+	}
+	cmd := &cobra.Command{
+		Use:     fmt.Sprintf("%s SOURCE_DIR POD:DESTINATION_DIR", name),
+		Short:   "Copy local files to a pod",
+		Long:    rsyncLong,
+		Example: fmt.Sprintf(rsyncExample, parent+" "+name),
+		Run: func(c *cobra.Command, args []string) {
+			kcmdutil.CheckErr(o.Complete(f, c, args))
+			kcmdutil.CheckErr(o.Validate())
+			kcmdutil.CheckErr(o.RunRsync())
+		},
+	}
+
+	cmd.Flags().StringVarP(&o.ContainerName, "container", "c", "", "Container within the pod")
+	cmd.Flags().BoolVarP(&o.Quiet, "quiet", "q", false, "Quiet copy")
```

Comment:
Put these two flags separated from other flags and add a comment "These flags must match rsync flag names"